pydoc-markdown/src/pydoc_markdown/contrib/renderers/markdown.py

```diff
@@ -205,7 +210,11 @@ def _render_toc(self, fp, level, obj):
     if level > self.toc_maxdepth:
       return
     object_id = self._resolver.generate_object_id(obj)
-    fp.write('  ' * level + '* [{}](#{})\n'.format(self._escape(obj.name), object_id))
+    title = self._escape(obj.name)
+    if not self.add_module_prefix and isinstance(obj, docspec.Module):
+      object_id = ".".join(object_id.split('.')[1:])
```

Comment:
The "object id" should not be affected by this new option. It's supposed to link to the anchor tags inserted by `insert_header_anchors`.